In [1]:
%%html
<style>
    table {display: inline-block}
</style>

# Summary Statistics

In [1]:
import numpy as np
import pandas as pd
import math
import statistics

In [2]:
# Brady Kids: Greg, 14; Marcia, 12; Peter, 11; Jan, 10; Bobby, 8; Cindy, 6; and Oliver, 8

# Create a list of the kids ages
ages = [14, 12, 11, 10, 8, 6, 8]

mean_age = np.mean(np.array(ages))
median_age = np.median(np.array(ages))
age_mode = statistics.mode(np.array(ages))
age_variance = statistics.variance(np.array(ages))
age_std_dev = statistics.stdev(np.array(ages))
age_std_err = age_std_dev/math.sqrt(len(ages))

output = f'''
The Brady children have an age mean of: {mean_age}, a Median age of: {median_age}, and the Mode of their ages is: {age_mode}.  
The children have an age Variance of: {age_variance}, with a Standard Deviation of: {age_std_dev}, and a Standard Error of: {age_std_err}.
'''

print(output)


The Brady children have an age mean of: 9.857142857142858, a Median age of: 10.0, and the Mode of their ages is: 8.  
The children have an age Variance of: 7, with a Standard Deviation of: 2.6457513110645907, and a Standard Error of: 1.0.



In [3]:
ages = [14, 12, 11, 10, 8, 7, 8]

mean_age = np.mean(np.array(ages))
median_age = np.median(np.array(ages))
age_mode = statistics.mode(np.array(ages))
age_variance = statistics.variance(np.array(ages))
age_std_dev = statistics.stdev(np.array(ages))
age_std_err = age_std_dev/math.sqrt(len(ages))

output = f'''
3) Post Birthday, the Brady children have an age mean of: {mean_age}, a Median age of: {median_age}, and the Mode of their ages is: {age_mode}.  
The children have an age Variance of: {age_variance}, with a Standard Deviation of: {age_std_dev}, and a Standard Error of: {age_std_err}.
'''

print(output)


3) Post Birthday, the Brady children have an age mean of: 10.0, a Median age of: 10.0, and the Mode of their ages is: 8.  
The children have an age Variance of: 6, with a Standard Deviation of: 2.449489742783178, and a Standard Error of: 0.9258200997725513.



In [4]:
ages = [14, 12, 11, 10, 8, 7, 1]

mean_age = np.mean(np.array(ages))
median_age = np.median(np.array(ages))

# Check for a lack of mode in the ages list
if len(set(ages)) != len(ages): 
    age_mode = statistics.mode(np.array(ages))
else: 
    age_mode = '''[ages] has no Mode'''
    
age_variance = statistics.variance(np.array(ages))
age_std_dev = statistics.stdev(np.array(ages))
age_std_err = age_std_dev/math.sqrt(len(ages))

output = f'''
After Oliver was replaced, the Brady children have an age mean of: {mean_age}, a Median age of: {median_age}, and the Mode of their 
ages is: {age_mode}.  The children have an age Variance of: {age_variance}, with a Standard Deviation of: {age_std_dev}, and a 
Standard Error of: {age_std_err}.
'''

print(output)


After Oliver was replaced, the Brady children have an age mean of: 9.0, a Median age of: 10.0, and the Mode of their 
ages is: [ages] has no Mode.  The children have an age Variance of: 18, with a Standard Deviation of: 4.242640687119285, and a 
Standard Error of: 1.6035674514745462.



<table>
    <caption style="border-style:solid;border-width: 0px 0px 1px 0px;">Summary table for Brady Kid scenerios</caption>
    <tr style="border-style:solid;border-width: 0px 0px 1px 0px;">
        <th></th>
        <th>Brady 1</th>
        <th>Brady 2</th>
        <th>Brady 3</th>
    </tr>
    <tr>
        <td>Mean $ \mu $</td>
        <td>9.8571</td>
        <td>10</td>
        <td>9</td>
    </tr>
    <tr>
        <td>Median</td>
        <td>10</td>
        <td>10</td>
        <td>10</td>
    </tr>
    <tr>
        <td>Mode</td>
        <td>8</td>
        <td>8</td>
        <td>$ \textit {No Mode} $</td>
    </tr>
    <tr>
        <td>Variance $ \sigma^2 $</td>
        <td>7</td>
        <td>6</td>
        <td>18</td>
    </tr>
    <tr>
        <td>Standard Deviation $ \sigma $</td>
        <td>2.6457</td>
        <td>2.4494</td>
        <td>4.2426</td>
    </tr>
    <tr style="border-style:solid;border-width: 0px 0px 1px 0px;">
        <td>Standard Error</td>
        <td>1.0</td>
        <td>.9258</td>
        <td>1.6035</td>
    </tr>
</table>
<h3>Assignments</h3>
<p>All responses reference the above table.</p>
<ol>
    <li>
        <p>The Brady kids have a Mean ($ \mu $) age of $ \textit{9.8571} $, a Median age of $ \textit{10} $ , and the Mode of their ages is $ \textit{8} $. </p>
        <p>The childeren have an age Variance ($ \sigma^2 $) of $ \textit{7} $, a Standard Deviation ($ \bar{x} $) of $ \textit{2.6457} $ and the Standard Error for the $ \textbf{Brady 1} $ scenerio is $ \textit{1} $.</p>
    </li>
    <li>
        <p>For scenerio $ \textbf{Brady 1} $ I believe that the Median is the representational value as a measure of Central Tendency.  When coupuled with Variance ($ \sigma^2 $) I can provide a sense of range of ages that are being dealt with.</p>
    </li>
    <li>
        <p>With the update to Cindy's age, in the $ \textbf{Brady 2} $ scenerio, the Mean ($ \bar{x} $) and Median converge on 10.  The age Variance ($ \sigma^2 $) drops by 1 year.  The Standard Deviation ($ \bar{x} $) and the Standard Error are both reduced.  This reflects the reduction in the age range.</p>
    </li>
    <li>
        <p>With scenerio $ \textbf{Brady 3} $ I would change to the Mean ($ \mu $) as being a better representation of the children.  I would also move off of Variance ($ \sigma^2 $) because of it's the substantial rise upon recalculation.</p>
    </li>
    <li>
        <p>This question is more intuitive than quantative.  TV Guide and Entertainment Weekly should have a broader audience than either Pop Culture Today or SciPhi Phanatic.  This should translate to their subscriber sample would be more representational of the overall population.  I'd be hesitant to say how predictive the percentages are since I don't have any details on where the numbers came from.  What is the sample size? What is the Standard Error?  Is there any Bias built into these percentages?  If directed to come up with an answer based solely on the information presented I say that $ \textit{21.5%} $ of Americans are fans of the Brady Bunch.  Where $ \textit{21.5%} $ is averaging the percentages from TV Guide and Entertainment Weekly.</p>
    </li>
</ol>